server/core/runtime/plan_step_runner.go

```diff
@@ -25,25 +25,29 @@ var (
 	minusDiffRegex = regexp.MustCompile(`(?m)^ {2}-`)
 )
 
-type PlanStepRunner struct {
+type planStepRunner struct {
 	TerraformExecutor   TerraformExec
 	DefaultTFVersion    *version.Version
 	CommitStatusUpdater StatusUpdater
 	AsyncTFExec         AsyncTFExec
 }
 
-func (p *PlanStepRunner) Run(ctx command.ProjectContext, extraArgs []string, path string, envs map[string]string) (string, error) {
+func NewPlanStepRunner(terraformExecutor TerraformExec, defaultTfVersion *version.Version, commitStatusUpdater StatusUpdater, asyncTFExec AsyncTFExec) Runner {
+	runner := &planStepRunner{
+		TerraformExecutor:   terraformExecutor,
+		DefaultTFVersion:    defaultTfVersion,
+		CommitStatusUpdater: commitStatusUpdater,
+		AsyncTFExec:         asyncTFExec,
+	}
+	return NewWorkspaceStepRunnerDelegate(terraformExecutor, defaultTfVersion, runner)
+}
+
+func (p *planStepRunner) Run(ctx command.ProjectContext, extraArgs []string, path string, envs map[string]string) (string, error) {
 	tfVersion := p.DefaultTFVersion
 	if ctx.TerraformVersion != nil {
 		tfVersion = ctx.TerraformVersion
 	}
 
-	// We only need to switch workspaces in version 0.9.*. In older versions,
-	// there is no such thing as a workspace so we don't need to do anything.
-	if err := p.switchWorkspace(ctx, path, tfVersion, envs); err != nil {
-		return "", err
-	}
```

Comment:
Why is this block removed?

Reply:
Th8s impl is moved into new delegation. Now plan strp ruunner also use it like followings. So removed.

https://github.com/runatlantis/atlantis/blob/7d12f313096f4001a11149085bb7594386ed282e/server/core/runtime/plan_step_runner.go#L42-L42